src/pages/index.js

```diff
@@ -64,50 +83,155 @@ profileEditPopup.setEventListeners();
 const imagePreviewPopup = new PopupWithImages("#card-picture-modal");
 imagePreviewPopup.setEventListeners();
 
+const cardDeletePopup = new PopupWithConfirmation("#delete-card-modal");
+cardDeletePopup.setEventListeners();
+
 const userInfo = new UserInfo({
   nameSelector: "#profile-title",
   jobSelector: "#profile-description",
+  pictureSelector: "#profile-image",
+});
+
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo({ name: userData.name, about: userData.about });
+  userInfo.setUserAvatar({ avatar: userData.avatar });
 });
 
 //Event listeners
 
 profileEditButton.addEventListener("click", function () {
-  profileTitleInput.value = profileTitle.textContent.trim();
-  profileDescriptionInput.value = profileDescription.textContent.trim();
-  editFormValidator.resetValidation();
   profileEditPopup.open();
 });
 
 cardAddButton.addEventListener("click", function () {
   newCardPopup.open();
 });
 
+// avatarPopup
+
+const avatarPopup = new PopupWithForm(
+  "#profile-picture-modal",
+  handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
+avatarEditButton.addEventListener("click", function () {
+  avatarPopup.open();
+});
+
 // Event Handlers
 
 function handleImageClick({ name, link }) {
   imagePreviewPopup.open({ name, link });
 }
 
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo({ name: data.name, job: data.description });
-  imagePreviewPopup.close();
+  profileEditPopup.setLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
got confused with this function in handleDeleteCardclick

Reply:
`handleSubmit` can be used inside `handleDeleteCardclick` as well like this:


```js
function handleDeleteCardclick(card) {
  cardDeletePopup.open();
  cardDeletePopup.setSubmitAction(() => {
    function makeRequest() {
      return api.removeCard(card._id).then(() => {
        card.handleDeleteCard();
        cardDeletePopup.close();
      });
    }
    handleSubmit(makeRequest, cardDeletePopup, "Deleting...");
  });
}

```